File: TrelloApi/TrelloApi/src/Domain/Entities/UserBoard.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;
    
[Table("UserBoard")]
public class UserBoard: Entity
{
    [ForeignKey("User")]
    public int UserId { get; set; }
    public User User { get; set; }
    
    [ForeignKey("Board")]
    public int BoardId { get; set; }
    public Board Board { get; set; }

    public UserBoard(int userId, int boardId)
    {
        UserId = userId;
        BoardId = boardId;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = null;
    }
}

File: TrelloApp/src/app/core/models/user-board.ts
import {User} from './user';
import {Board} from './board';

export interface UserBoard {
  id: number,
  createdAt: string,
  updatedAt?: string,
  boardId: number,
  userId: number,
  user: User,
  board: Board
}

export interface AddUserBoard {
  userId: number,
}
